src/segment.rs

```diff
@@ -250,6 +253,48 @@ impl<'a> Segment<'a> {
             func(Segment { inner: None })
         }
     }
+
+    ///
+    /// Create an external segment within this transaction.
+    ///
+    /// Example:
+    ///
+    /// ```rust
+    /// use std::{thread, time::Duration};
+    ///
+    /// use newrelic::{App, ExternalParamsBuilder};
+    ///
+    /// # if false {
+    /// let app = App::new("Test app", "Test license key")
+    ///     .expect("Could not create app");
+    /// let transaction = app
+    ///     .web_transaction("Test transaction")
+    ///     .expect("Could not start transaction");
+    /// let segment_params = ExternalParamsBuilder::new("https://www.rust-lang.org/")
+    ///     .procedure("GET")
+    ///     .library("reqwest")
+    ///     .build()
+    ///     .expect("Invalid external segment parameters");
+    /// {
+    ///     let segment = transaction.create_external_segment(&segment_params);
+    ///     let _headers = segment.distributed_trace()?;
+    ///     thread::sleep(Duration::from_secs(1))
+    /// }
+    /// # }
```

Comment:
```suggestion
    /// # }
    /// # Ok::<(), newrelic::Error>(())
```

This doctest fails without this due to the `?`.

Reply:
Ah I can see you changed this to remove the `?`. Since users will need to use the headers perhaps it'd be better to re-add the `?`, and add an `    /// # Ok::<(), newrelic::Error>(())` below the existing doctest lines, which will be a bit more representative of real usage. (There's an example of this in the [documentation tests](https://doc.rust-lang.org/rustdoc/documentation-tests.html#using--in-doc-tests) Rust docs.

Reply:
Huh.  I thought it had changed this.  Any objection to taking out the `if false` on this so that doesn't happen again?

Reply:
No objection - the tests don't all currently pass without the New Relic key anyway. Hopefully I'll get CI working soon...

Reply:
I verified it's working now.

I had also previously changed the distributed_trace call not to return a result since, it shouldn't ever fail to decode a base64 string as utf8.